hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java

```diff
@@ -571,23 +571,37 @@ public ServerName getRegionServerOfRegion(RegionInfo regionInfo) {
    * wants to iterate this exported list.  We need to synchronize on regions
    * since all access to this.servers is under a lock on this.regions.
    *
-   * @return A clone of current assignments.
+   * @return A clone of current open or opening assignments.
    */
   public Map<TableName, Map<ServerName, List<RegionInfo>>> getAssignmentsForBalancer(
       TableStateManager tableStateManager, List<ServerName> onlineServers) {
     final Map<TableName, Map<ServerName, List<RegionInfo>>> result = new HashMap<>();
     for (RegionStateNode node : regionsMap.values()) {
+      // DisableTableProcedure first sets the table state to DISABLED and then force unassigns
+      // the regions in a loop. The balancer should ignore all regions for tables in DISABLED
+      // state because even if still currently open we expect them to be offlined very soon.
       if (isTableDisabled(tableStateManager, node.getTable())) {
+        if (LOG.isTraceEnabled()) {
+          LOG.trace("Ignoring " + node + " because table is disabled");
```

Comment:
nit: good to provide placeholder: `LOG.trace("Ignoring {} because table is disabled", node);`

Reply:
I don't agree with this nit but I'll do it

Reply:
Is one better than the other? I'd say no because it's easy to edit the string and not change the arguments with printf style log lines. I've done that, I've seen others do it. For your future consideration.